citar.el

```diff
@@ -834,15 +833,24 @@ With prefix, rebuild the cache before offering candidates."
   (bibtex-completion-insert-bibtex
    (citar--extract-keys keys-entries)))
 
+(defun citar--insert-bibtex (key)
+  "Insert the bibtex entry for KEY at point."
+  ;; TODO: this doesn't work correctly.
+  (with-temp-buffer
+    (citar--open-entry key)
+    (bibtex-copy-entry-as-kill))
+  (bibtex-yank))
```

Comment:
Haven't tested but I'd suggest something like this, which avoids going via the kill ring.

~~~ elisp
(let ((beg (bibtex-beginning-of-entry))
      (end (bibtex-end-of-entry)))
  (buffer-substring-no-properties beg end))
~~~

Reply:
Right, the second option does not work.

Reply:
The first one does correctly insert the entry, but then also (I think via `bibtex-find-entry` opens the bibtex file with entry at point.

I thought the `with-temp-buffer` would avoid that?

Reply:
Okay, this needs more thought :)

Reply:
Hmm, [bibtex-completion](https://github.com/tmalsburg/helm-bibtex/blob/aa775340ba691d2322948bfdc6a88158568a1399/bibtex-completion.el#L683) uses a regex search and parsebib to do this.

Maybe `bibtex-find-entry` is the wrong tool here?

Reply:
I think I know the issue. The GLOBAL arg for `bibtex-find-entry` will search the buffer only when nil.

So I think could insert the file contents in the temp-buffer and use that?

Reply:
This works:

```elisp
(defun citar--insert-bibtex (key)
  "Insert the bibtex entry for KEY at point."
  (let ((bibtex-files
         (seq-concatenate 'list citar-bibliography (citar--local-files-to-cache))))
    (insert
     (with-temp-buffer
       (dolist (bib-file bibtex-files)
         (insert-file-contents bib-file))
       (bibtex-find-entry key)
       (let ((beg (bibtex-beginning-of-entry))
             (end (bibtex-end-of-entry)))
         (buffer-substring-no-properties beg end))))))
```